File: 0167_Two_Sum_II_Input_Array_Is_Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if numbers is None or len(numbers) < 2:
            return [-1, -1]

        left = 0
        right = len(numbers) - 1

        while left < right:
            tmp = numbers[left] + numbers[right]
            if tmp > target:
                right -= 1
            elif tmp < target:
                left += 1
            else:
                return [left + 1, right + 1]

        return [-1, -1]

File: cpp/0167_Two_Sum_II-Input_Array_Is_Sorted.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {        
        sort(nums.begin(), nums.end());
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int cur = nums[l] + nums[r];
            if (cur == target) {
                return {l + 1, r + 1};
            } else if (cur < target) {
                l++;
            } else {
                r--;
            }
        }
        return {};
    }
};